.github/actions/setup-node/action.yml

```diff
@@ -0,0 +1,64 @@
+name: "Setup Node Environment"
```

Comment:
I don't think this whole workflow is actually needed? the default `setup-node` action already manages a cache for the node_modules folder since v3 (as described at https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request) 

So I think we can greatly simplify the workflow?

Reply:
Oh I missed this that in V3 it caches the node_modules.